main.js

```diff
@@ -107,6 +171,32 @@ class eWeLink {
 
     return { status: 'ok', state };
   }
+
+  async toggleDevice(deviceId, channel = 1) {
+    const powerState = await this.getDevicePowerState(deviceId, channel);
+    const state = _get(powerState, 'state', false);
+
+    if (!state) {
+      return { error: powerState.error, msg: device.msg | 'Device does not exist' };
+    }
+
+    const newState = state === 'on' ? 'off' : 'on';
+
+    const newResponse = await this.setDevicePowerState(
+      deviceId,
+      newState,
+      channel
+    );
+
+    return newResponse;
+  }
+
+  async getDeviceChannelCount(deviceId) {
+    const device = await this.getDevice(deviceId);
+    const switchesAmount = getDeviceChannelCount(device);
```

Comment:
i've renamed this method to not get confused by this and the one on helpers file